### PR TITLE
support cloning gitlab's CI_JOB_TOKEN

### PR DESCRIPTION
**Pull request by saucoide**
Gitlab CI runs have a $CI_JOB_TOKEN (https://docs.gitlab.com/ee/ci/jobs/ci_job_token.html) which needs a different username vs using personal token, which can reuse the format used for github's auth

this only adds an extra case for when CI_JOB_TOKEN exists

### CODE REVIEW

**Comment by jdoiro3** [comment]
@saucoide, can you elaborate more on the running on gitlab and relying on a local gif config? I'll like to understand that part before merging.

**Comment by jdoiro3** [comment]
You'll also need to add this in this file too. Hoping to eliminate this duplication at some point but for now we'll need to keep these two files in sync.

https://github.com/jdoiro3/mkdocs-multirepo-plugin/blob/3ce10854592f70537fced28bc86b74b41628f4c9/mkdocs_multirepo_plugin/scripts/sparse_clone_old.sh#L19-L26

**Comment by saucoide** [comment]
@jdoiro3 sure will add

`CI_JOB_TOKEN` is gitlab's default, it's always there, so if we use that variable name, we'd be making using it the default behavior, if anyone had already set it up in gitlab to work in a different way*, we'd be breaking it since we don't fall using only `$url`, now i convinced myself a different env var would be better, but whatever you prefer :)

*for example someone's CI might be writing the token & configuring git authentication as part of the CI job just before calling mkdocs so that it'd use that token, or doing something like this https://github.com/jdoiro3/mkdocs-multirepo-plugin/issues/33#issuecomment-1176079194

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "mkdocs-multirepo-plugin"
-version = "0.7.0"
+version = "0.7.1"
```
Comment (jdoiro3) [inline pyproject.toml:3]: Can you revert this change? I'm going to bump the version and generate a new release after this is merged. I'd like to include other changes. Once you revert this I'll merge. Thanks for contributing by the way!

Comment (saucoide) [reply pyproject.toml]: @jdoiro3 done, thanks :)

**Pull request merged**